File: Assignment_43/Program1.c
/*
Write a recursive program which accept string from user and count white spaces
Input : HE llo WOr lD
Output : 3
*/

#include <stdio.h>

int WhiteSpace(char *str)
{
    static int iCount = 0;

    if(*str != '\0')
    {
       if(*str == ' ')
        {
            iCount++;
        }
        str++;
        WhiteSpace(str);
    }
    return iCount;
}

int main()
{
    int iRet = 0;
    char arr[20] = {'\0'};

    printf("Enter string : ");
    scanf("%[^\n]",arr);

    iRet = WhiteSpace(arr);

    printf("%d",iRet);

    return 0;
}

File: Assignment_55/Program1.java
/* Write a java program which accept array of characters from user & replace each capital character with its corresponding small charcater.
Input :   b N j B R b A d G G
Output :  b n j b r b a d g g
*/

import java.util.*;

class MyArray
{
    public void ArrayReplace(char arr[])
    {
        int i = 0;

        for(i = 0;i < arr.length;i++)
        {
            if(Character.isUpperCase(arr[i]))
            {
                arr[i] = Character.toLowerCase(arr[i]);
            }
        }

        System.out.println("Replaced capital characters Array :");

        for(char ch : arr)
        {
            System.out.print(ch+"\t");
        }
        System.out.println();
    }
}

class Program1
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int i = 0;

        System.out.println("Enter characters in the array :");
        String line = sobj.nextLine();
        String[] tokens = line.trim().split("\\s+");
        int size = tokens.length;
        char arr[] = new char[size];
        for (i = 0;i < size;i++)
        {
            arr[i] = arr[i] = tokens[i].charAt(0);
        }

        MyArray mobj = new MyArray();

        mobj.ArrayReplace(arr);

        sobj.close();
    }
}